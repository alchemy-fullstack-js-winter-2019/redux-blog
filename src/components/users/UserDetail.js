import React from 'react';
import PropTypes from 'prop-types';
import Post from '../posts/Post';

export default function UserDetail({ user, posts }) {
  const listOfPosts = posts.map((post, i) => {
    return (
      <li key={i}>
        <Post post={post} />
      </li>
    );
  });
  return (
    <>
      <h2>{user.name}</h2>
      <p>{user.email}</p>
      <ul>{listOfPosts}</ul>
    </>
  );
}

UserDetail.propTypes = {
  user: PropTypes.object,
  posts: PropTypes.array
};
