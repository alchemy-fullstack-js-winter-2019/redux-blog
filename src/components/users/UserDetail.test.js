import React from 'react';
import renderer from 'react-test-renderer';
import UserDetail from './UserDetail';
import { MemoryRouter } from 'react-router-dom';

describe('UserDetail Snapshot', () => {
  it('matches a snapshot', () => {
    const user = {
      'id': 1,
      'name': 'Leanne Graham',
      'username': 'Bret',
      'email': 'Sincere@april.biz',
      'address': {
        'street': 'Kulas Light',
        'suite': 'Apt. 556',
        'city': 'Gwenborough',
        'zipcode': '92998-3874',
        'geo': {
          'lat': '-37.3159',
          'lng': '81.1496'
        }
      },
      'phone': '1-770-736-8031 x56442',
      'website': 'hildegard.org',
      'company': {
        'name': 'Romaguera-Crona',
        'catchPhrase': 'Multi-layered client-server neural-net',
        'bs': 'harness real-time e-markets'
      }
    };
    const tree = renderer.create(
      <MemoryRouter>
        <UserDetail user={user} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
