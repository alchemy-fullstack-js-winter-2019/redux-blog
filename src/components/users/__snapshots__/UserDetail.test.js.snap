// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail matches a snapshot 1`] = `
Array [
  <h2>
    name
  </h2>,
  <p>
    email@email
  </p>,
  <ul>
    <li>
      <a
        href="/posts/undefined"
        onClick={[Function]}
      >
        <h2>
          title1
        </h2>
      </a>
      <p>
        body1
      </p>
    </li>
    <li>
      <a
        href="/posts/undefined"
        onClick={[Function]}
      >
        <h2>
          title2
        </h2>
      </a>
      <p>
        body2
      </p>
    </li>
  </ul>,
]
`;
